packages/dexter/src/governance_admin.rs

```diff
@@ -0,0 +1,272 @@
+    /// As governance admin contract is designed to become owner of Vault and Multi-staking contracts, any configuration change on them can be done using this message
+    /// This message can be executed only by the Governace module of the chain i.e. it is fully chain governed.
+    ExecuteMsgs { msgs: Vec<CosmosMsg> },
+
+    /// Resumes the pool creation process after the proposal is passed.
+    /// This message is called by the governance module of the chain as an action of the proposal created by the `CreatePoolCreationProposal` message.
+    /// This message can only be called by the governance module of the chain.
+    ResumeCreatePool { pool_creation_request_id: u64 },
+
+    /// Resumes the reward schedule creation process.
+    /// This message is called by the governance module of the chain as an action of the proposal created by the `CreateRewardSchedulesProposal` message.
+    /// This message can also be called by the contract itself after a new pool with reward schedules is created using the `ResumeCreatePool` message.
+    ResumeCreateRewardSchedules {
+        reward_schedules_creation_request_id: u64,
+    },
+
+    // Self executable
+    /// Callback message to store the proposal id after the proposal is created.
+    /// This message is ordered by the `CreatePoolCreationProposal` message after the `SubmitProposal` message, so it is executed right after the proposal is created in the same transaction.
+    /// This message is executed by the governance admin contract itself and no-one else can execute this message.
+    PostGovernanceProposalCreationCallback {
+        gov_proposal_type: GovAdminProposalRequestType,
+    },
+
+    /// Callback message after the pool is created in the vault.
+    /// This message is ordered by the `ResumeCreatePool` message after the `CreatePool` message, so it is executed right after the pool is created in the same transaction.
+    /// This message is executed by the governance admin contract itself and no-one else can execute this message.
+    /// We trigger the pool join functionality and store pool specific data in the Gov admin contract state in this message.
+    ResumeJoinPool { pool_creation_request_id: u64 },
+}
+
+#[cw_serde]
+#[derive(QueryResponses)]
+pub enum QueryMsg {
+    /// Returns the state of the pool creation request
+    #[returns(PoolCreateRequestContextData)]
+    PoolCreationRequest { pool_creation_request_id: u64 },
+
+    /// Returns the state of the reward schedule creation request
+    #[returns(RewardScheduleCreationRequestsState)]
+    RewardScheduleRequest { reward_schedule_request_id: u64 },
+
+    /// Returns the refundable funds for the user.
+    /// It provides total refund and also a breakdown of the refundable funds so that the user can understand the reason for the refund
+    /// and the calculation involved in the refund.
+    #[returns(RefundResponse)]
+    RefundableFunds {
+        request_type: GovAdminProposalRequestType,
```

Comment:
Add a query message for getting the funds needed for PoolCreation and Reward schedule creation without having to query it on frontend.

Reply:
Done